### DISCUSSION TITLE
Unreal Engine 5.1

### DISCUSSION THREAD

**Discussion by chrixbedardcad** [microdee/OpenUnrealSpaceMouse#82] [General]
The Space Mouse plugin on Unreal Marketplace is not available for the latest version of Unreal Engine 5.1. Any plan to make that happen soon ? What is the option at the moment, to recompile it ourselves for UE5.1?

I got the following when I try to install **[SpaceMouse-Win64-1.2.5.5.0.3-PreBuilt.zip](https://github.com/microdee/UE4-SpaceMouse/releases/download/1.2.5/SpaceMouse-Win64-1.2.5.4.24.3-PreBuilt.zip)**

<img width="645" alt="SpaceMouseError" src="https://user-images.githubusercontent.com/75350983/199558358-83f75364-9074-44cc-8fd4-7fef7ea04cd3.png">

**Comment by microdee** [comment]
5.1 is in preview yet, it's not released as a stable engine, and you cannot submit your plugins to Marketplace for preview versions (max version I can select in the dashboard is 5.0)
Currently your options are building it yourself https://github.com/microdee/UE4-SpaceMouse-TestProject#build-spacemouse-for-unreal-engine or wait for me to pre-compile for 5.1, or wait until 5.1 is properly released by Epic.

**Reply by chrixbedardcad** [reply]
Ok, thanks for the reply. Do you plan to pre-compile for 5.1 soon and share it in GitHub? 

Note: Meanwhile, I will try to build it myself as you suggest, but downloading 5.1 for compilation is a killer...

**Comment by anttin2020** [comment]
This plugin is absolutely amazing, I have been using the 3Dconnexion plugin and it only works for viewport, not even cameras.
But I cannot live without my Spacepilot Pro in Unreal and I might start experimenting with 5.1 but without this plugin (3DC plugin does not work either) it is pain :)

I checked the instructions to build my own version but already Nuke looks very complex and I have practically no experience with git in development.

So if you or anyone here could make the plugin prebuilt for UE 5.1, I (and I'm sure many others) would be VERY happy :)

BTW, I assume it is not possible to make the LCD button help to work with this?

Anyway, thank you very much for your excellent work!

**Reply by microdee** [reply]
https://github.com/microdee/UE4-SpaceMouse/releases/download/1.2.5/SpaceMouse-Win64-1.2.5.5.1.0-PreBuilt.zip here's the release on github, I'll upload to the marketplace shortly

> Nuke looks very complex

Nah it's not, but Nuke is optional and it's making your life easier if you're a C++ dev or a devops engineer, by expressing the following with simple command line syntax, but you can compile it with regular means: 

1. Clone the test project recursively with git
2. generate project files
3. open Visual Studio
4. Build Solution with "Development Editor" configuration and "Win64" platform is selected
5. Open project as regular
6. Open Plugins window
7. Package plugin to be used by other projects
8. You can install the packaged plugin to the engine then

> BTW, I assume it is not possible to make the LCD button help to work with this?

It would be really cool, it crossed my mind for the SpaceMouse Enterprise, but that would require me to capture and reverse engineer the USB packets sent to the device by the official driver, which is a lot harder to see of course than the other way around.

**Comment by chrixbedardcad** [comment]
Hey, Unreal Engine 5.1 is now officially released, I hope we will get the 3D Mouse plugin soon :-P

**Reply by microdee** [reply]
oh cool I'll submit an update then today

**Reply by microdee** [reply]
https://github.com/microdee/UE4-SpaceMouse/releases/download/1.2.5/SpaceMouse-Win64-1.2.5.5.1.0-PreBuilt.zip you can try this, I'll update marketplace soon

**Reply by chrixbedardcad** [reply]
Yes, I just test it, and works like a charm!! It feels so good to get back the 3D Mouse working in UE5.1!
Thank you so much for this quick update!
